src/librustc/traits/mod.rs

```diff
@@ -237,8 +237,8 @@ pub enum ObligationCauseCode<'tcx> {
     /// Computing common supertype in the arms of a match expression
     MatchExpressionArm(Box<MatchExpressionArmCause<'tcx>>),
 
-    /// Computing common supertype in the pattern guard for the arms of a match expression
-    MatchExpressionArmPattern { span: Span, ty: Ty<'tcx> },
+    /// Computing common supertype in a pattern guard
+    PatternGuard(Box<PatternGuardCause<'tcx>>),
```

Comment:
`PatternGuard` feels misleading as the usage in `demand_eqtype_pat_diag` is about patterns. (There's also no such thing as a *pattern guard*. Only arms have guards.)